auth0/v3/management/rest.py

```diff
@@ -105,8 +105,8 @@ def delete(self, url, params={}):
     def _process_response(self, response):
         text = json.loads(response.text) if response.text else {}
 
-        if isinstance(text, dict) and 'errorCode' in text:
+        if isinstance(text, dict) and (response.status_code is None or response.status_code >= 400):
             raise Auth0Error(status_code=text['statusCode'],
-                             error_code=text['errorCode'],
+                             error_code=text.get('errorCode', ''),
```

Comment:
Another PR suggests falling back to the property "error" when "errorCode" is not present in the response.

Reply:
I've looked at the latest change to `/authentication/base.py` and it mostly looks good because it both takes `status_code` into account and uses `error` instead of `errorCode`.  The only issue I noticed is that it is not python 2 compatible.  Using `super().__init__(...)` is not supported in python 2.  To be supported in both python 2 and python 3 this should be `super(Response, self).__init__(...)`

Reply:
I will update my PR with this change in `/authentication/base.py` and update `/management/rest.py` to use the same logic.

Reply:
Got it @beck3905, thanks. @darkyen can you please give me a hand reviewing the python side of things here? I don't even know which is the minimum python version we support. I guess from [here](https://github.com/auth0/auth0-python/blob/master/setup.py#L32) that we target 2.7 and up.